compiler/src/dmd/expressionsem.d

```diff
@@ -10012,17 +10012,22 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             (ae.e2.type.ty == Tsarray || ae.e2.type.ty == Tarray) &&
             (ae.e1.type.nextOf && ae.e2.type.nextOf && ae.e1.type.nextOf.mutableOf.equals(ae.e2.type.nextOf.mutableOf));
 
-        if (!isArrayAssign)
+        const isArraySetAssign =
+            (ae.e1.isSliceExp || ae.e1.type.ty == Tsarray) &&
+            (ae.e1.type.nextOf && ae.e1.type.nextOf.mutableOf.equals(ae.e2.type));
```

Comment:
Should this be checking if e2.type implicitly converts to e1.type.nextOf rather than equals?